src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Annotations.cs

```diff
@@ -26,6 +27,14 @@ public IPythonType GetTypeFromAnnotation(Expression expr, LookupOptions options
             switch (expr) {
                 case null:
                     return null;
+                case NameExpression nameExpr:
```

Comment:
Can it also be a `MemberExpression` i.e. `bound=A.B` such as inner class?

Reply:
This is for generic self, these are annotations, basically checking when an expression is annoted `a: T` that it returns the annotation is generic, as it wasn't doing that beforehand.